auth0/auth0_client.go

```diff
@@ -4,18 +4,54 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
-
 	"github.com/parnurzeal/gorequest"
+	"net/http"
 )
 
 type AuthClient struct {
 	config *Config
 }
 
-func NewClient(config *Config) *AuthClient {
+func NewClient(clientId string, clientSecret string, config *Config) (*AuthClient, error) {
+
+	token, err := getToken(clientId, clientSecret, config.domain, config.apiUri)
+
+	if err != nil {
+		return nil, fmt.Errorf("auth0 provider init failed, error: %v", err)
+	}
+
+	config.accessToken = token
+
 	return &AuthClient{
 		config: config,
+	}, nil
+}
+
+func getToken(clientId string, clientSecret string, domain string, apiUri string) (string, error) {
+	auth0LoginRequest := &LoginRequest{
+		ClientId:     clientId,
+		ClientSecret: clientSecret,
+		Audience:     apiUri,
+		GrantType:    "client_credentials",
 	}
+
+	res, body, errs := gorequest.New().Post("https://" + domain + "/oauth/token").Send(auth0LoginRequest).End()
```

Comment:
Maybe worth adding a `Retry` here too just in case, since some users of this project may have Free tenants.

![image](https://user-images.githubusercontent.com/13288003/66054387-cbf9a200-e52b-11e9-89f3-29e79010b6fb.png)

Reply:
This function is executed only once per provider so we can expect one request per provider creation. Is `each` provider initialised only once per terraform `plan/apply`? 

I wouldn't worry too much about `rate limiting` here.

But as it is an easy change there is no harm in adding a retry on `429` here.

Reply:
I saw weird behaviour in testing where the provider was configured multiple times per test case, but I think that is an edge case (users will not run it like this unless they are contributing). I just worry about the potential confusion in large companies using this - for example maybe one developer is load testing Auth0 integrations, while another is attempting to use this provider. 😄